stackit/internal/services/objectstorage/credentialsgroup/resource.go

```diff
@@ -0,0 +1,340 @@
+			"Error importing credentialsGroup",
+			fmt.Sprintf("Expected import identifier with format [project_id],[credentials_group_id], got %q", req.ID),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("credentials_group_id"), idParts[1])...)
+	tflog.Info(ctx, "ObjectStorage credentials group state imported")
+}
+
+func mapFields(credentialsGroupResp *objectstorage.CreateCredentialsGroupResponse, model *Model) error {
+	if credentialsGroupResp == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if credentialsGroupResp.CredentialsGroup == nil {
+		return fmt.Errorf("response credentialsGroup is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	credentialsGroup := credentialsGroupResp.CredentialsGroup
+
+	model.Name = types.StringPointerValue(credentialsGroup.DisplayName)
+	model.URN = types.StringPointerValue(credentialsGroup.Urn)
+
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.CredentialsGroupId.ValueString(),
+	}
+	model.Id = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	return nil
+}
+
+// enableProject enables object storage for the specified project. If the project already exists, nothing happens
+func (r *credentialsGroupResource) enableProject(ctx context.Context, model *Model) error {
+	projectId := model.ProjectId.ValueString()
+
+	// From the object storage OAS: Creation will also be successful if the project already exists, but will not create a duplicate
+	_, err := r.client.CreateProject(ctx, projectId).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create object storage project: %w", err)
+	}
+	return nil
+}
+
+// readCredentialsGroups gets all the existing credentials groups for the specified project,
+// finds the credentials group that is being read and updates the state. If the credentials group cannot be found, it throws an error
+func readCredentialsGroups(ctx context.Context, model *Model, projectId string, client *objectstorage.APIClient) error {
+	found := false
+
+	if model.CredentialsGroupId.ValueString() == "" && model.Name.ValueString() == "" {
+		return fmt.Errorf("missing configuration: either name or credentials group id must be provided")
+	}
+
+	credentialsGroupsResp, err := client.GetCredentialsGroups(ctx, projectId).Execute()
+	if err != nil {
+		return fmt.Errorf("getting credentials groups: %w", err)
+	}
+
+	if credentialsGroupsResp == nil {
+		return fmt.Errorf("nil response from GET credentials groups")
+	}
+
+	for _, credentialsGroup := range *credentialsGroupsResp.CredentialsGroups {
+		if *credentialsGroup.CredentialsGroupId != model.CredentialsGroupId.ValueString() && *credentialsGroup.DisplayName != model.Name.ValueString() {
+			continue
+		}
+		found = true
+		model.CredentialsGroupId = types.StringValue(*credentialsGroup.CredentialsGroupId)
```

Comment:
Does it make sense to modify `mapFields` so you can call it here? It would make sense, since this is basically the same logic (I think). Maybe have `mapFields` get `objectstorage.CredentialsGroup` as input?

Reply:
done